flocker/control/schema/types.yml

```diff
@@ -140,12 +140,33 @@ definitions:
       "^.+$":
         type: string
 
+  volume:
+    title: "A volume attached to a container"
+    description: |
+      A dataset that has been mounted inside a container.
+    type: object
+    properties:
+      dataset_id:
+        '$ref': '#/definitions/dataset_id'
+      mountpoint:
+        '$ref': '#/definitions/volume_path'
+    required:
+      - 'dataset_id'
+      - 'mountpoint'
+    additionalProperties: false
+
+  volume_path:
+    title: "Volume mount path"
+    description: |
+      The filesystem path where a volume is mounted inside a container.
+    type: string
+
```

Comment:
Is there a pattern this could be constrained to for better validation? I'm guessing it has to be an absolute path.
`^((?:\/[a-zA-Z0-9]+(?:_[a-zA-Z0-9]+)*(?:\-[a-zA-Z0-9]+)*)+)$` :smile: